### DISCUSSION TITLE
The configuration file stores passwords in plaintext, which may be considered insecure by ICSL. Is there any solution?

### DISCUSSION THREAD

**Discussion by chinahnwqf** [strongswan/strongswan#2545] [Q&A]
The configuration file stores passwords in plaintext, which may be considered insecure by ICSL. Is there any solution?
![image](https://github.com/user-attachments/assets/65dd8093-7f70-47fc-9538-df9d8d031566)

**Comment by tobiasbrunner** [comment]
Not really. At some point some secret has to be available to the daemon in plaintext to use pre-shared keys (whether it's the actual PSK or e.g. a key to decrypt it via a custom plugin doesn't really matter). You might want to consider using certificates and store private keys on smart cards (you still need a PIN to unlock them, but they won't be stored/available directly).

**Reply by chinahnwqf** [reply]
I was wondering if this could pass in ciphertext, decrypt it in open source, and assign it to variables.

**Reply by tobiasbrunner** [reply]
Sorry, I don't understand what you mean.

**Reply by chinahnwqf** [reply]
Currently, plaintext is stored in the file. I want to store ciphertext in the file. Then when strongswan uses the key, I take it out and decrypt the ciphertext. This is safer, equivalent to plaintext being used in memory variables, not directly in files. (In the file, many people can see it with root permission.)

**Reply by chinahnwqf** [reply]
Do you think it's okay?

**Reply by tobiasbrunner** [reply]
How does it decrypt it? Where does that key come from?

**Reply by chinahnwqf** [reply]
We first agree on a common encryption algorithm, according to this storage, strongswan according to the agreement to decrypt.

**Reply by tobiasbrunner** [reply]
Again, where does the key come from to decrypt it (the algorithm is irrelevant)? Where is it stored?
